drivers/display/display_st7528i.c

```diff
@@ -0,0 +1,513 @@
+}
+
+static int st7528i_write_palette(const struct device *dev)
+{
+	int ret;
+
+	ret = st7528i_set_mode(dev, ST7528_MODE_PALETTE);
+	if (ret) {
+		return ret;
+	}
+
+	for (int index = 0; index < 16; index++) {
+		uint8_t value = ST7528_PWM_LEVELS - (index * ST7528_PWM_LEVELS / 15);
+		uint8_t cmd = ST7528_CMD_SET_PALETTE_ENTRY | (index << 2);
+
+		/* same value for all four "frames" - not entirely clear from the datasheet
+		 * what that even means
+		 */
+		uint8_t cmd_buf[] = {
+			cmd | 0x00, value, cmd | 0x01, value, cmd | 0x02, value, cmd | 0x03, value,
+		};
+
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	ret = st7528i_set_mode(dev, ST7528_MODE_NORMAL);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7528i_reset(const struct device *dev)
+{
+	const struct st7528i_config *config = dev->config;
+	int ret;
+
+	if (config->reset.port) {
+		gpio_pin_set_dt(&config->reset, true);
+		k_sleep(ST7528_RESET_DELAY);
+		gpio_pin_set_dt(&config->reset, false);
+		k_sleep(ST7528_RESET_DELAY);
+	}
+
+	{
+		uint8_t cmd_buf[] = {
+			ST7528_CMD_SET_PARTIAL_DISPLAY_LINES,
+			config->com_partial_display,
+			ST7528_CMD_ADC_SELECT | config->invert_segments,
+			ST7528_CMD_SHL_SELECT | (config->invert_com << 3),
+			ST7528_CMD_SET_INITIAL_COM0,
+			config->com_offset,
+
+			ST7528_CMD_START_OSCILLATOR,
+			ST7528_CMD_SET_REGULATOR_RESISTOR | config->regulator_resistor,
+			ST7528_CMD_SET_ELECTRONIC_VOLUME,
+			config->electronic_volume,
+			ST7528_CMD_SET_LCD_BIAS | (config->lcd_bias - 5),
+
+			ST7528_CMD_SET_BOOST | 0x00,
+		};
+
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	k_sleep(K_MSEC(200));
+
+	{
+		uint8_t cmd_buf[] = {
+			ST7528_CMD_POWER_CONTROL | ST7528_POWER_CONTROL_VC,
+			ST7528_CMD_SET_BOOST | (config->boost - 3),
+		};
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	k_sleep(K_MSEC(200));
+
+	{
+		uint8_t cmd_buf[] = {
+			ST7528_CMD_POWER_CONTROL | ST7528_POWER_CONTROL_VC |
+				ST7528_POWER_CONTROL_VR,
+		};
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	k_sleep(K_MSEC(10));
+
+	{
+		uint8_t cmd_buf[] = {
+			ST7528_CMD_POWER_CONTROL | ST7528_POWER_CONTROL_VC |
+				ST7528_POWER_CONTROL_VR | ST7528_POWER_CONTROL_VF,
+			ST7528_CMD_SET_FRC_PWM | ST7528_FRC_4 |
+				CONCAT(ST7528_PWM_, ST7528_PWM_LEVELS),
+
+		};
+		ret = st7528i_write_cmds(dev, cmd_buf, ARRAY_SIZE(cmd_buf));
+		if (ret) {
+			return ret;
+		}
+	}
+
+	ret = st7528i_write_palette(dev);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7528i_init(const struct device *dev)
+{
+	const struct st7528i_config *config = dev->config;
+	int ret;
+
+	if (config->height != ST7528_HEIGHT) {
+		LOG_ERR("Invalid display height: expected %d, got %d", ST7528_HEIGHT,
+			config->height);
```

Comment:
> if it is possible to wrap the line at an argument [...] so that the preferred line length is respected

What do you mean by that? The line you commented on *is* wrapped at an argument and the line length is respected. Can you elaborate on the style you would prefer?